src/Qir/Runtime/lib/Simulators/FullstateSimulator.cpp

```diff
@@ -408,7 +419,109 @@ namespace Quantum
 
             return (std::abs(actualProbability - probabilityOfZero) < precision);
         }
-    };
+
+      private:
+        std::ostream& GetOutStream(const void* location, std::ofstream& outFileStream);
+        void DumpMachineImpl(std::ostream& outStream);
+        void DumpRegisterImpl(std::ostream& outStream, const QirArray* qubits);
+        void GetStateTo(TDumpLocation location, TDumpToLocationCallback callback);
+        bool GetRegisterTo(TDumpLocation location, TDumpToLocationCallback callback, const QirArray* qubits);

```

Comment:
Critically, this private method does not get a register, but the quantum state of that register (this is a has-a versus is-a distinction, in that a register has a state, but a register is not a state). We should make absolutely sure to keep that distinction clear in our documentation and our naming conventions.

Reply:
I'm not sure what you mean. The parameter `qubits` is a pointer to `QirArray` containing the qubit _ids_ (not qubit _values_). 
Do you mean that I need to rename `qubits` to `qubitIds` or something like that?
Please elaborate, feel free to give me a Teams call.